### DISCUSSION TITLE
Sort by for hierarchical menu with custom data

### DISCUSSION THREAD

**Discussion by jornwildenbeest** [algolia/instantsearch#5121] [Q&A]
Hello,

i'm trying to sort my hierarchical menu with custom data from algolia.

When I use a custom sortBy function i only get the following data returned to work with:

```
count:  217
data: null
escapedValue: "Kasten"
exhaustive: true
isRefined: false
name: "Some name"
path: "Some path"
```

How do I add custom data to this object?

**Comment by Haroenv** [comment]
you can add custom data to that object in `transformItems`, which gets called before `sortBy`. You can also sort directly in that function

**Reply by jornwildenbeest** [reply]
unfortunately transformItems does not work either, the data set is the same. What i want is adding custom data that I inserted into Algolia in these functions.

**Reply by Haroenv** [reply]
Unfortunately, facets can only be strings, they don't contain the whole record that had that facet

**Reply by jornwildenbeest** [reply]
So at the moment there is no way of achieving this?

**Reply by Haroenv** [reply]
You can normalise the data inside the facet itself, depending on what it is, `categories > something _ postfix` and removing the `_ postfix` from the search response before returning the search response to InstantSearch, but this would require features in the engine to work correctly

**Reply by jornwildenbeest** [reply]
Not sure what you mean by this.

Could you explain this? Removing the `_ postfix` from the search response will make the data return in the sort function?

**Reply by Haroenv** [reply]
you can store the other data in a separate key of the response and then map it in transformItems. Sorry that this isn't straightforward, but I haven't seen anyone try this in production yet, so I don't know what works best
